client.go

```diff
@@ -80,6 +80,10 @@ type Client struct {
 	protocol    Protocol
 	logger      hclog.Logger
 	doneCtx     context.Context
+
+	// NegotiatedVersion is a record of the protocol version that was returned
+	// by the server.
+	NegotiatedVersion int
```

Comment:
It looks like this becomes valid after `Start` returns, and would always be 0 (effectively garbage) before that. Maybe worth mentioning that in the doc comment so it's easier to understand how to properly use this.

Reply:
SGTM :+1: (mitchellh just mentioned the same thing ;) )